lib/grell/page.rb

```diff
@@ -224,37 +216,56 @@ def all_links
         []
       end
 
-      # We only accept links in this same host that start with a path
-      # nil from this
-      def link_to_url(link)
-        uri = URI.parse(link)
-        if uri.absolute?
-          if uri.host != URI.parse(host).host
-            Grell.logger.debug "GRELL does not follow links to external hosts: #{link}"
-            nil
-          else
-            link # Absolute link to our own host
-          end
-        else
-          if uri.path.nil?
-            Grell.logger.debug "GRELL does not follow links without a path: #{uri}"
-            nil
-          end
-          if uri.path.start_with?('/')
-            host + link  #convert to full URL
-          else #links like href="google.com" the browser would go to http://google.com like "http://#{link}"
-            Grell.logger.debug "GRELL Bad formatted link: #{link}, assuming external"
-            nil
-          end
+      # Private class to group all the methods related to links.
+      class Link
+        def initialize(anchor)
+          @anchor = anchor
         end
 
-      rescue URI::InvalidURIError #We will have invalid links propagating till we navigate to them
-        link
-      end
-    end
+        # <link> can only be used in the <head> as of: https://developer.mozilla.org/en/docs/Web/HTML/Element/link
+        def inside_header?
+          @anchor.tag_name == 'link'
+        end
 
+        # Is the link disabled by either Javascript or CSS?
+        def disabled?
+          @anchor.disabled? || !!@anchor.native.attributes['disabled']
+        end
 
+        # Some links may use data-href + javascript to do interesting things
+        def href
+          @anchor['href'] || @anchor['data-href']
+        end
 
-  end
+        # We only accept links in this same host that start with a path
+        # nil from this
+        def to_url(host)
+          uri = URI.parse(href)
+          if uri.absolute?
+            if uri.host != URI.parse(host).host
+              Grell.logger.debug "GRELL does not follow links to external hosts: #{href}"
+              nil
+            else
+              href # Absolute link to our own host
+            end
+          else
+            if uri.path.nil?
+              Grell.logger.debug "GRELL does not follow links without a path: #{uri}"
+              nil
+            end
+            if uri.path.start_with?('/')
+              host + href  #convert to full URL
```

Comment:
I know it's copy paste code, but there's a bunch of comments missing a space between the # and the text